support/cas-server-support-audit-core/src/main/java/org/apereo/cas/audit/AbstractAuditTrailManager.java

```diff
@@ -0,0 +1,50 @@
+package org.apereo.cas.audit;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.inspektr.audit.AuditActionContext;
+
+import java.time.LocalDate;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+/**
+ * This is {@link AbstractAuditTrailManager}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.2.0
+ */
+@Slf4j
+@Setter
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+public abstract class AbstractAuditTrailManager implements GenericAuditTrailManager {
```

Comment:
This is excellent. I recommend you remove the interface altogether and instead simply modify the signature of `getAuditRecordsSince` to be more generic. It doesn't make sense to have both methods in there.

Reply:
I can't modify the signature of `getAuditRecordsSince` here because the interface is not defined in CAS.

Reply:
Ah, gotcha. Let me work that into inspektr then, and cut a release, or would you rather do that as well?

Reply:
Done. 
PR apereo/inspektr#12

Reply:
Wonderful. Lets get this merged then based on the current inspektr snapshot and all things pass, I'll cut a quick release. 

Thank you.